.github/actions/setup-rust/tests/conftest.py

```
@@ -0,0 +1,36 @@
+    for parent in (start, *start.parents):
+        if (parent / "pyproject.toml").exists() or (parent / ".git").exists():
+            return parent
+    raise FileNotFoundError(str(start))
+
+
+ROOT = _find_root(Path(__file__).resolve())
+sys.path.insert(0, str(ROOT))
+
+
+@pytest.fixture
+def shell_stubs(tmp_path: Path, monkeypatch: pytest.MonkeyPatch):
```

Ruff (0.12.2)

Missing return type annotation for public function `shell_stubs`

(ANN201)